src/KOKKOS/fix_rigid_kokkos.cpp

```diff
@@ -0,0 +1,1274 @@
+  memoryKK->create_kokkos(k_fflag,fflag,nbody,3,"rigid/kk:fflag");
+  memoryKK->create_kokkos(k_tflag,tflag,nbody,3,"rigid/kk:tflag");
+
+  
+  // The initialization for these two is simple enough:
+  for (int i = 0; i < nbody; i++) {
+    k_fflag.d_view(i,0) = k_fflag.d_view(i,1) = k_fflag.d_view(i,2) = 1.0;
+    k_tflag.d_view(i,0) = k_tflag.d_view(i,1) = k_tflag.d_view(i,2) = 1.0;
+    if (domain->dimension == 2) k_fflag.d_view(i,2) = k_tflag.d_view(i,2) = 0.0;
+  }
+  k_fflag.template modify<LMPDeviceType>();
+  k_tflag.template modify<LMPDeviceType>();
+
+  // These seem fine.
+  memoryKK->create_kokkos(k_omega, omega, nbody,3,"rigid/kk:omega");
+  memoryKK->create_kokkos(k_angmom,angmom,nbody,3,"rigid/kk:angmom");
+
+  
+  // These are allocated in the base c-tor but set later.
+  // Specifically, quat, ex_space, ey_space, ez_space and inertia
+  // are set in setup_bodies_static() which is called in init().
+  memoryKK->create_kokkos(k_quat,   quat,    nbody, 4,"rigid/kk:quat");
+  memoryKK->create_kokkos(k_inertia,inertia, nbody, 3,"rigid/kk:inertia");
+
+  if (debug_output && comm->me == 0) {
+    fprintf(stderr, "Contents of body:");
+    for (int i = 0; i < nbody; ++i) {
+      fprintf(stderr, " %i", body[i]);
+    }
+    fprintf(stderr, "\n");
+  }
+
+  int nmax = atomKK->nmax;
+
+  // eflags and k_body are already set in the c-tor of FixRigid:
+  k_eflags.modify<LMPHostType>();
+  k_body.modify<LMPHostType>();
+  memoryKK->create_kokkos(k_eflags, eflags, nmax, "rigid/kk:eflags");
+  memoryKK->create_kokkos(k_body,   body,   nmax,  "rigid/kk:body");
+  
+  if (debug_output && comm->me == 0) {
+    fprintf(stderr, "Contents of body:");
+    for (int i = 0; i < nbody; ++i) {
+      fprintf(stderr, " %i", body[i]);
+    }
+    fprintf(stderr, "\n");
+  }
+
+  memoryKK->create_kokkos(k_ex_space,ex_space, nbody, 3,"rigid/kk:ex_space");
+  memoryKK->create_kokkos(k_ey_space,ey_space, nbody, 3,"rigid/kk:ey_space");
+  memoryKK->create_kokkos(k_ez_space,ez_space, nbody, 3,"rigid/kk:ez_space");
+
+  // displace, body, xcmimage and eflags are the size of nmax.
+  memoryKK->create_kokkos(k_displace, displace, nmax, 3,"rigid/kk:displace");
+
+  if (debug_output && comm->me == 0) {
+    fprintf(stderr, "Using fix rigid/kokkos!\n");
+  }
+}
+
+template <class DeviceType>
+FixRigidKokkos<DeviceType>::~FixRigidKokkos()
+{
+  memoryKK->destroy_kokkos(k_xcm);
+  memoryKK->destroy_kokkos(k_vcm);
+  memoryKK->destroy_kokkos(k_fcm);
+
+  memoryKK->destroy_kokkos(k_tflag);
+  memoryKK->destroy_kokkos(k_fflag);
+
+  memoryKK->destroy_kokkos(k_omega);
+  memoryKK->destroy_kokkos(k_angmom);
+  memoryKK->destroy_kokkos(k_torque);
+
+  memoryKK->destroy_kokkos(k_quat);
+  memoryKK->destroy_kokkos(k_inertia);
+
+  memoryKK->destroy_kokkos(k_ex_space);
+  memoryKK->destroy_kokkos(k_ey_space);
+  memoryKK->destroy_kokkos(k_ez_space);
+
+  memoryKK->destroy_kokkos(k_displace);
+}
+
+
+
+template <class DeviceType>
+void FixRigidKokkos<DeviceType>::cleanup_copy()
```

Comment:
This is not needed when using lambda functions.

Reply:
OK, great!